gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -168,4 +170,32 @@ public static Writable deserializeFromString(Class<? extends Writable> writableC
       closer.close();
     }
   }
+
+  /**
+   * Serializes a given {@link FsPermission} object into a {@link String} and stores it as a key, value pair in the
+   * given {@link State}. This method is meant to be used in conjunction with
+   * {@link #deserializeFsPermission(State, String)}.
+   */
+  public static void serializeFsPermission(State state, String key, FsPermission fsPermissions) {
```

Comment:
Do you want to expose the key to the clients? This client 'may' misuse the key to store something else.